### PR TITLE
fix: support subpath in remappings

### LINKED ISSUE #1137: brownie-config.yaml remappings bug
### Environment information

* `brownie` Version: 1.14.6
* `solc` Version: 0.6.6
* Python Version: 3.8.10
* OS: ubuntu

```yaml
dependencies:
  - Uniswap/Uniswap-v2-core@1.0.0
  - Uniswap/Uniswap-lib@2.1.0


compiler:
  solc:
    version: 0.6.6
    remappings:
      - "@uniswapcore=Uniswap/Uniswap-v2-core@1.0.0"
      - "@uniswaplib=**Uniswap**/Uniswap-lib@2.1.0"
```

```
brownie compile 

contracts/UniswapV2Migrator.sol:3:1: ParserError: Source "Uniswap/Uniswap-lib@2.1.0/contracts/libraries/TransferHelper.sol" not found: File not found.
import '@uniswaplib/contracts/libraries/TransferHelper.sol'
```

```
make .brownie/packages/Uniswaplib
mv  Uniswap/Uniswap-lib@2.1.0 Uniswaplib
```

```yaml
compiler:
  solc:
    version: 0.6.6
    remappings:
      - "@uniswapcore=Uniswap/Uniswap-v2-core@1.0.0"
      - "@uniswaplib=**Uniswaplib**/Uniswap-lib@2.1.0"
```

brownie compile  OK

### PR DESCRIPTION
**Pull request by kalloc**
### What I did

Fixes #1137

### How I did it

Add support for mappings to subfolders

### How to verify it

```yaml
compiler:
    solc:
        version: null
        optimizer:
            enabled: true
            runs: 31337
        remappings:
            - "@openzeppelin=OpenZeppelin/openzeppelin-contracts@3.4.0"
            - "@openzeppelin/contracts-upgradeable=OpenZeppelin/openzeppelin-contracts-upgradeable@3.4.0/contracts"
```

### Checklist

- [x] I have confirmed that my PR passes all linting checks
- [ ] I have included test cases
- [ ] I have updated the documentation
- [x] I have added an entry to the changelog

### CODE REVIEW

File: brownie/project/compiler/__init__.py [hunk 1]
```diff
@@ -14,9 +14,7 @@
 from brownie.exceptions import UnsupportedLanguage
 from brownie.project import sources
 from brownie.project.compiler.solidity import (  # NOQA: F401
-    find_best_solc_version,
     find_solc_versions,
-    install_solc,
```
Comment (iamdefinitelyahuman) [inline brownie/project/compiler/__init__.py:19]: removing these imports is breaking other functionality

Comment (kalloc) [reply brownie/project/compiler/__init__.py]: sorry, it was my mistake, fixed

File: brownie/project/compiler/__init__.py [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+import solcast
 
 from brownie._config import _get_data_folder
 from brownie.exceptions import UnsupportedLanguage
 from brownie.project import sources
-from brownie.project.compiler.solidity import (  # NOQA: F401
+from brownie.project.compiler.solidity import (
     find_best_solc_version,
     find_solc_versions,
     install_solc,
```
Comment (iamdefinitelyahuman) [inline brownie/project/compiler/__init__.py:19]: still failing the lint CI..  i think everything that's changed in this section needs to be undone.

Comment (kalloc) [reply brownie/project/compiler/__init__.py]: Yeah, my mistake, returned it back

Comment (iamdefinitelyahuman) [reply brownie/project/compiler/__init__.py]: still need to change some of this back to how it was.  you might consider running `tox -e lint` locally to check everything.

**Review by iamdefinitelyahuman — APPROVED** [review]

**Comment by iamdefinitelyahuman** [comment]
Closes #1137

**Pull request merged**